### PR TITLE
Separate status and event listeners

### PR DESCRIPTION
**Pull request by akikanellis**
@dadouf

### CODE REVIEW

File: sdk/src/main/java/com/supenta/flitchio/sdk/FlitchioController.java [hunk 1]
```diff
@@ -121,15 +123,19 @@ public void onServiceConnected(ComponentName className, IBinder service) {
...
                     FlitchioLog.e("Unexpected error: could not identify this controller.");
                 }
             }
 
             // We connect the client in case he asked for it while binding was not ready. The client
-            // should not call onResume() if its listener is null, but in case he did, we don't
+            // should not call onResume() if its eventListener is null, but in case he did, we don't
             // register the client for a null Listener.
-            if (activityLifecycleMoment == ActivityLifecycle.ON_RESUME && listener != null) {
+            if (activityLifecycleMoment == ActivityLifecycle.ON_RESUME && eventListener != null) {
```
Comment (dadouf) [inline sdk/src/main/java/com/supenta/flitchio/sdk/FlitchioController.java]: Here the condition should include the statusListener as well. I think:
`if (activityLifecycleMoment == ActivityLifecycle.ON_RESUME && (eventListener != null || statusListener != null))`

File: sdk/src/main/java/com/supenta/flitchio/sdk/FlitchioController.java [hunk 2]
```diff
@@ -292,39 +298,37 @@ public boolean onCreate() throws FlitchioManagerDependencyException {
...
+     * a {@link FlitchioEventListener}.
      *
-     * @param listener The listener.
-     * @param handler  The handler associated to the thread on which the callbacks will happen.
+     * @param eventListener The eventListener.
+     * @param handler       The handler associated to the thread on which the callbacks will happen.
      * @since 0.5.0
      */
-    public void onResume(FlitchioListener listener, Handler handler) {
+    public void onResume(FlitchioEventListener eventListener, FlitchioStatusListener statusListener, Handler handler) {
```
Comment (dadouf) [inline sdk/src/main/java/com/supenta/flitchio/sdk/FlitchioController.java]: It is a detail, but considering that StatusListener is likely to be more frequently used than EventListener (I think), I'd put it as first parameter.

**Comment by dadouf** [comment]
I think that -- apart from my two comments -- the separation Status/Event is fine, so you can move on two the next two steps before this branch is ready for merging:
- broadcast receiving and notification of Status event up to the listener
- Javadoc (as last step :p)

**Comment by akikanellis** [comment]
@dadouf I have done the fixes (and some Javadoc from before but I will finalize it) but I can't understand your first bullet-point. Didn't we decide to not use a broadcast but the normal way of how the API works except with the two listeners?

**Comment by akikanellis** [comment]
Javadoc will come after we finalize the code ;)

**Comment by akikanellis** [comment]
Yes you are right.

**Comment by akikanellis** [comment]
@dadouf I tried to follow the naming conventions on the manager but it was overcomplicated for this usage so I have reverted to something simpler.

**Comment by dadouf** [comment]
All the code looks fine to me now!

**Comment by dadouf** [comment]
:+1: now to release 0.6.0!

**Pull request merged**